src/snowflake/cli/api/project/schemas/entities/snowpark_entity.py

```diff
@@ -82,13 +82,25 @@ def convert_runtime(cls, runtime_input: Union[str, float]) -> str:
             return str(runtime_input)
         return runtime_input
 
-    @field_validator("artifacts")
-    @classmethod
-    def validate_artifacts(cls, artifacts: List[Path]) -> List[Path]:
-        for artefact in artifacts:
-            if "*" in str(artefact):
-                raise ValueError("Glob patterns not supported for Snowpark artifacts.")
-        return artifacts
+    def get_artifacts(self) -> List[PathMapping]:
```

Comment:
NADE's path globbing logic is centralized in BundleMap. Any particular reason we can't reuse it here instead of doing a one-off implementation? I'm worried that different glob expansion implementations will end up making things confusing for customers since the specific globbing semantics can diverge over time (and reading what's below, might have diverged already TBH).

Reply:
It's intended to be generic, but feel free to take a look and tell me what you think.